precompile/contracts/warp/contract_warp_handler.go

```diff
@@ -69,16 +69,17 @@ func handleWarpMessage(accessibleState contract.AccessibleState, input []byte, s
 
 	// Note: we charge for the size of the message during both predicate verification and each time the message is read during
 	// EVM execution because each execution incurs an additional read cost.
-	msgBytesGas, overflow := math.SafeMul(GasCostPerWarpMessageBytes, uint64(len(predicateBytes)))
+	// Compute underlying bytes length as len(pred)*32 (padded size).
+	paddedLen := uint64(len(pred)) * uint64(common.HashLength)
+	msgBytesGas, overflow := math.SafeMul(GasCostPerWarpMessageBytes, paddedLen)
 	if overflow {
 		return nil, 0, vm.ErrOutOfGas
 	}
 	if remainingGas, err = contract.DeductGas(remainingGas, msgBytesGas); err != nil {
 		return nil, 0, err
 	}
-	// Note: since the predicate is verified in advance of execution, the precompile should not
-	// hit an error during execution.
-	unpackedPredicateBytes, err := predicate.UnpackPredicate(predicateBytes)
+	// Inline decode using AvalancheGo predicate API
```

Comment:
I think the prior comment is still correct and more meaningful.
```suggestion
	// Note: since the predicate is verified in advance of execution, the precompile should not
	// hit an error during execution.
```